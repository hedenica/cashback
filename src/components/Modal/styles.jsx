import styled, { keyframes } from 'styled-components';
import { X } from 'react-feather';

export const Background = styled.div`
  display: flex;
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  justify-content: center;
  align-self: center;
  background: rgba(85, 85, 85, 0.8);
  z-index: 2;
`;

const appearFromTop = keyframes`
  from {
    opacity: 0;
    transform: translateY(-100%);
  }
  to {
    opacity: 1;
    transform: translateY(0%);
  }
`;

export const ModalWrapper = styled.div`
  width: 800px;
  height: 500px;
  box-shadow: 0 5px 16px rgba(0, 0, 0, 0.2);
  background: #f7f7f7;
  display: grid;
  grid-template-columns: 1fr 1fr;
  position: absolute;
  z-index: 10;
  border-radius: 10px;
  margin: 200px auto;

  animation: ${appearFromTop} 1s;

  @media (max-width: 767px) {
    display: flex;
    position: fixed;
    width: 100%;
    height: 100%;
    flex-direction: column;
    bottom: 0;
    border-radius: 0;
    margin: auto;
  }
`;

export const CloseModal = styled(X)`
  cursor: pointer;
  position: absolute;
  top: 20px;
  right: 20px;
  width: 32px;
  height: 32px;
  padding: 0;
  z-index: 10;

  &:hover {
    color: #fbab8c;
  }
`;
