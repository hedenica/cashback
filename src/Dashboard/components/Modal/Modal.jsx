/* eslint-disable no-unused-vars */
import React, { useRef } from 'react';
import PropTypes from 'prop-types';
import { useSpring, animated } from 'react-spring';

import { Background, ModalWrapper, CloseModal } from './styles';

const Modal = ({ isOpen, onClose, firstModalContent, secondModalContent }) => {
  const modalRef = useRef(null);

  const animation = useSpring({
    config: {
      duration: 250,
    },
    opacity: isOpen ? 1 : 0,
    transform: isOpen ? `translateY(0%)` : `translateY(-100%)`,
  });

  const closeModal = e => {
    if (modalRef.current === e.target) {
      onClose();
    }
  };

  return (
    <>
      {isOpen && (
        <Background ref={modalRef} onClick={closeModal}>
          <animated.div style={animation}>
            <ModalWrapper>
              {firstModalContent}
              {secondModalContent}
              <CloseModal aria-label="Fechar modal" onClick={onClose} />
            </ModalWrapper>
          </animated.div>
        </Background>
      )}
    </>
  );
};

export default Modal;

Modal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  firstModalContent: PropTypes.node,
  secondModalContent: PropTypes.node.isRequired,
};

Modal.defaultProps = {
  firstModalContent: '',
};
