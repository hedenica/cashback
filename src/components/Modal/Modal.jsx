import React, { useRef } from 'react';
import PropTypes from 'prop-types';

import { Background, ModalWrapper, CloseModal } from './styles';

const Modal = ({ isOpen, onClose, firstModalContent, secondModalContent }) => {
  const modalRef = useRef(null);

  const closeModal = e => {
    if (modalRef.current === e.target) {
      onClose();
    }
  };

  return (
    <>
      {isOpen && (
        <Background ref={modalRef} onClick={closeModal}>
          <ModalWrapper>
            {firstModalContent}
            {secondModalContent}
            <CloseModal aria-label="Fechar modal" onClick={onClose} />
          </ModalWrapper>
        </Background>
      )}
    </>
  );
};

export default Modal;

Modal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  firstModalContent: PropTypes.node,
  secondModalContent: PropTypes.node.isRequired,
};

Modal.defaultProps = {
  firstModalContent: '',
};
