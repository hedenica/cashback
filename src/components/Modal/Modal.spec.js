import React from 'react';
import { render } from '@testing-library/react'

import Modal from './Modal'

describe('<Modal />', () => {
  it('should return the modal component', () => {
    const isOpen = true 

    const { container } = render(
      <Modal 
        isOpen
        onClose={() => !isOpen}
        firstModalContent="Lado esquerdo do Modal"
        secondModalContent="Lado direito do Modal"
      />
    )
    expect(container).toMatchSnapshot()
  })
})
