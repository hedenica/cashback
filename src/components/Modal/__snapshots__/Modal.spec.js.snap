// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should return the modal component 1`] = `
<div>
  <div
    class="sc-bdfBwQ cCeiBc"
  >
    <div
      class="sc-gsTCUz lceZVE"
    >
      Lado esquerdo do Modal
      Lado direito do Modal
      <svg
        aria-label="Fechar modal"
        class="sc-dlfnbm biQaJe"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="18"
          x2="6"
          y1="6"
          y2="18"
        />
        <line
          x1="6"
          x2="18"
          y1="6"
          y2="18"
        />
      </svg>
    </div>
  </div>
</div>
`;
